src/castra/middleware.clj

```diff
@@ -1,69 +1,55 @@
 (ns castra.middleware
-  (:require
-    [clojure.java.shell             :as sh]
-    [cognitect.transit              :as t]
-    [ring.middleware.session.cookie :as c]
-    [clojure.string                 :as string]
-    [ring.util.request              :as q :refer [body-string]]
-    [clojure.set                    :as s :refer [intersection difference]]
-    [castra.core                    :as r :refer [ex ex? dfl-ex *pre* *request* *session* *validate-only*]]
-    [clojure.stacktrace             :as u :refer [print-cause-trace print-stack-trace]])
-  (:import
-    [java.io File]
-    [java.util.regex Pattern]
-    [java.io ByteArrayInputStream ByteArrayOutputStream]))
-
-;; hat tip to io.aviso/pretty
-(def ^:const ^:private ansi-pattern (Pattern/compile "\\e\\[.*?m"))
-(defn ^String strip-ansi
-  "Removes ANSI codes from a string, returning just the raw text."
-  [string]
-  (string/replace string ansi-pattern ""))
+  (:require [cognitect.transit :as t]
+            [ring.util.request :as q :refer [body-string]]
+            [clojure.set       :as s :refer [intersection difference]]
+            [castra.core       :as r :refer [ex? dfl-ex *pre* *request*
+                                             *session* *validate-only*]])
+  (:import [java.io ByteArrayInputStream ByteArrayOutputStream]))
 
 (defn- ex->clj [e]
   (let [e (if (ex? e) e (dfl-ex e))]
     {:message (.getMessage e)
-     :data    (ex-data e)
-     ;; Commenting out `:stack`
-     ;; This is required for security compliance:
-     ;;
-     ;; - This information might help an attacker gain more information and potentially
-     ;;   focus on the development of further attacks for the target system.
-     ;;
-     ;;- An attacker can understand internal workflow by analyzing stack traces.
-     ;;
-     ;;- The stack trace can disclose potentially sensitive information such as:
-     ;;    - physical file paths of relevant files
-     ;;    - source code fragments
-     ;;    - version information of various packages
-     ;;    - database information
-     ;;    - error messages
-     ;;
-     ;; Please see [sc-28285] for more context
-     ;; TODO: Consider adding an option to conditionally add this for Adzerk Admins
-     #_#_:stack   (strip-ansi
-                   (with-out-str
-                     (try (print-cause-trace e)
-                          (catch Throwable x
-                            (try (print-stack-trace e)
-                                 (catch Throwable x
-                                   (printf "No stack trace: %s" (.getMessage x))))))))}))
```

Comment:
Removed this commented-out chunk of code. It seems clear to me at this point that we will never want to send stacktraces from the server to the client, even for Kevel admins. When running the UI locally, we can already see the stacktraces in the console. And we have a good logging setup in Datadog that allows us to see the stacktraces.